miio/gateway/sub_device/sub_device.py

```diff
@@ -0,0 +1,221 @@
+"""Xiaomi Gateway subdevice base class."""
```

Comment:
Maybe following could be a better directory structure for subdevices?
```
gateway/gateway.py
gateway/alarm.py
gateway/devices/subdevice.py (this main class)
gateway/devices/switch.py
```
etc.

Reply:
agreed

Reply:
done